### ISSUE TITLE
1.2.5 Necessary Hotfixes

### ISSUE THREAD

**Issue by ClassyTeddy** [CCDevNelson/Snap-Updates#331]
I am creating this thread so that we can log any issues that need to be put into a 1.2.5 hotfix. I will create a list of issues and add to that list as more things are found. 

- [ ] Need to re add dog ear prompts to the carcass construction options

@AntonioGloria, I heard that there is a problem with counting the number of shelves? Please comment more on that issue so I can add it to the list.

**Comment by ClassyTeddy** [comment]
The dog ear prompts should be available again on the `1.2.5-Hotfixes` branch.

**Comment by AntonioGloria** [comment]
Hi @ClassyTeddy,
Yes, the issue seems to be specifically with the Shelf Stacks assembly. It wasn't taking into account the assembly's cutparts' visibility when counting them. I'm working on that fix.

**Comment by AntonioGloria** [comment]
I have just pushed the fix for the Shelf Stacks shelf count bug.

**Comment by TylerStandage** [comment]
@AntonioGloria @ClassyTeddy thanks guys, glad to see the fixes were easy.  Antonio is going to dig into the issue of pards showing up double in 2D views when rendering a second time over the next biz day or two.  According to Katie the dog ear pards are used, but not a ton, so let's let designers and other users have another day or two before we send out a hotfix.

@sarahhightower if this plan sounds bad to you, let us know and we'll try to get it out sooner than later.

**Comment by sarahhightower** [comment]
Sounds good to me!

**Comment by AntonioGloria** [comment]
I noticed another issue. With some entry door types, the return wall isn't being created with a break in the plan views. I can look into this one as well
![image_2021_04_23T20_25_19_680Z](https://user-images.githubusercontent.com/74732175/115935306-87f52780-a458-11eb-8158-f3c85e1b3ce3.png)

**Comment by TylerStandage** [comment]
good catch, @AntonioGloria .  please do look into it.  I think we'll shoot for a Tuesday hotfix release.

**Comment by AntonioGloria** [comment]
I have pushed the fix for the return wall breaks at door frames:
![image](https://user-images.githubusercontent.com/74732175/116106746-0380ef80-a678-11eb-869c-29c6940c5c21.png)

**Comment by sarahhightower** [comment]
Per Teddy, posting issues I noticed today while in training.

- Glass Shelves not working within door prompt
- Floor missing on elevation
- Dust shelves being given incorrect depth

**Comment by ClassyTeddy** [comment]
> Dust shelves being given incorrect depth

@sarahhightower Which dust shelves that are apart of what assembly? Is it wrong just in the scene, the 2D views, or both? How wrong is the depth and what should it be?

**Comment by sarahhightower** [comment]
I've only noticed it on the dust shelves in the standard double hang. The depth should be 11.75 and they are showing as 18". The dust shelves look correct in the scene and incorrect in the 2D. I've discussed this with @AntonioGloria as well. 

![Screenshot 2021-04-26 151517](https://user-images.githubusercontent.com/72047269/116157665-3b9d2800-a6a2-11eb-8e23-b8afcc266333.jpg)

**Comment by ClassyTeddy** [comment]
> Glass Shelves not working within door prompt

This is now working and is ready for testing in the `1.2.5-Hotfixes` branch. I have moved the glass shelves prompts over to the "Shelf Options" tab.

**Comment by AntonioGloria** [comment]
> I've only noticed it on the dust shelves in the standard double hang. The depth should be 11.75 and they are showing as 18". The dust shelves look correct in the scene and incorrect in the 2D. I've discussed this with @AntonioGloria as well.

I'm not sure how the wrong depth dimensions got there, but just now I tried removing the labels and re-adding them and they didn't come up again. However, I got double parding. I'll keep looking into it.

![image](https://user-images.githubusercontent.com/74732175/116159909-b2442080-a6b6-11eb-99f2-fc087477185a.png)

**Comment by TylerStandage** [comment]
Thanks all, seems this will go another day before we can get the hot release out. Thanks for your work on this and talk to you tomorrow!

**Comment by AntonioGloria** [comment]
I just pushed a fix to clear orphan data and remove double-parding.

**Comment by TylerStandage** [comment]
Thanks Antonio! I  believe this branch now has all the updates expected for this hot update. Sarah is going to do some testing as will I in a bit. Anyone else that wishes to do some testing, please do and report any issues you've found. If all looks good several hours by now, we'll request @RyanMDev  make the release!

**Comment by AntonioGloria** [comment]
I have pushed a fix for some errors with 2D Views to the `1.2.5-Hotfixes` branch:
- In some cases, when generating 2D Views more than once in a single session, some assemblies or other parts of the room - for example, the floor - would go missing.
- While clearing orphan data helped with removing double-parding in some cases, it was only part of the solution. I have added a definitive solution to double-parding.
